src/main/java/datawave/microservice/query/cachedresults/CachedResultsQueryService.java

```diff
@@ -0,0 +1,1362 @@
+                    QueryLogicFactory queryLogicFactory, QueryStorageCache queryStorageCache, ResponseObjectFactory responseObjectFactory,
+                    MarkingFunctions markingFunctions, CachedResultsQueryParameters cachedResultsQueryParameters) {
+        this.cachedResultsQueryProperties = cachedResultsQueryProperties;
+        this.cachedResultsJdbcTemplate = cachedResultsJdbcTemplate;
+        this.cachedResultsQueryCache = cachedResultsQueryCache;
+        this.queryService = queryService;
+        this.auditClient = auditClient;
+        this.securityMarking = securityMarking;
+        this.queryLogicFactory = queryLogicFactory;
+        this.queryStorageCache = queryStorageCache;
+        this.responseObjectFactory = responseObjectFactory;
+        this.markingFunctions = markingFunctions;
+        this.cachedResultsQueryParameters = cachedResultsQueryParameters;
+        this.fieldDefinitions = IntStream.range(0, cachedResultsQueryProperties.getNumFields()).mapToObj(x -> FIELD + x + " LONGTEXT")
+                        .collect(Collectors.joining(", "));
+        this.preparedFields = IntStream.range(0, cachedResultsQueryProperties.getNumFields()).mapToObj(x -> FIELD + x).collect(Collectors.joining(", "));
+        this.preparedValues = Stream.generate(() -> "?").limit(cachedResultsQueryProperties.getNumFields()).collect(Collectors.joining(", "));
+        initializeTableTemplate();
+    }
+    
+    private void initializeTableTemplate() {
+        // @formatter:off
+        String createTableTemplate = cachedResultsQueryProperties.getStatementTemplates().getCreateTableTemplate()
+                .replace(FIELD_DEFINITIONS_PLACEHOLDER, fieldDefinitions);
+        // @formatter:on
+        try {
+            cachedResultsJdbcTemplate.execute(createTableTemplate);
+        } catch (DataAccessException e) {
+            log.error("Unable to create 'create table' template with statement: {}", createTableTemplate, e);
+            throw e;
+        }
+    }
+    
+    // Prior to the LOAD call, the user defines a query via the query API
+    
+    // TODO: JWO: Add a method to manually remove a query's results from the cache
+    
+    // TODO: JWO: Should this call stay open forever, or should we return immediately?
+    // I think the previous code stayed open forever.
+    /**
+     * Creates the specified query, runs it to completion, and caches the results in SQL
+     *
+     * The query represented by queryId will be duplicated, and it's results will be written to SQL.
+     * 
+     * @param definedQueryId
+     *            the query id of the defined query, not null
+     * @param alias
+     *            an optional, user-defined alias which can be used to reference the query
+     * @param currentUser
+     *            the current user, not null
+     * @return the view name
```

Comment:
put in @throws annotation for javadoc too

Reply:
sure